wordpress-mobile/WordPress-Android
# Jetpack Focus: Enhance screenshots

Parent #16900

This PR includes:

- New methods added to `JPScreenshotTest` to generate the following screenshots
-- Edit blog post
-- Notifications
-- Media
-- Blogging Reminders
-- Blogging Prompts
-- Choose site topic (launched via Site Creation)
-- Create new options (post, story, page) - bottom sheet shown after FAB click
-- Choose a layout (after clicking new page)
- A `Screenshots` enum was introduced to manage which screenshots are enabled for generation. Before generation a check is made to validate if the screenshot is enabled. 
- Includes method name refactoring
- Includes reorganization of methods within the class
- Adds a few helpers to support running the screenshots in any order
- Fixes the `EditBlog` screenshot within WPScreenshotTest mentioned here #16846 
- Add mocks for blogging prompts, layouts and site topics

**Notes**
Not all screenshots mentioned above are enabled for generation, but the logic exists to generate them (minus some content in the mocks). 

To test:
1. Checkout the PR branch locally 
2. Switch to `jetpackJalapenoDebug` build variant.
3. Run the `jPScreenshotTest` from `JPScreenshotTest` in Android Studio to check that it **succeeds**.

   ⚠️ **Note:** _When running multiple times you may get fails with Hilt or sometimes wiremock. We are starting an investigation into upgrading our wiremock library (we are really, really, really, far behind). Hopefully that will fix the wiremock issues._

   <details>
   <summary>Emulator Setup Guide</summary>

     Ideally we run the tests on an emulator created by fastlane. See the notes in the issue for how this can be done:

     > Follow the guide in `SCREENSHOT_DEVICE_SETUP.md` to create a device like the ones we're using for screenshots automation (this won't be super easy as the guide is a bit outdated and if you're on an M1 Mac you have to change the generated .ini files manually to set the arm64 architecture or else the emultators won't start.
   </details>

4. Close emulator & wipe data
5. Switch to `wordpressJalapenoDebug` build flavor.
6. Run the `wPScreenshotTest` from `WPScreenshotTest` to check that it **succeeds**, meaning we're regression free.


## Regression Notes
1. Potential unintended areas of impact
WP Screenshot tests no longer work

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Ran WPScreenshot tests and validated they function as expected

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
